platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/FileSigningUtils.java

```diff
@@ -0,0 +1,402 @@
+     */
+    @NonNull
+    private static byte[] signData(@NonNull final byte[] data, @NonNull final KeyPair keyPair)
+            throws NoSuchAlgorithmException, NoSuchProviderException, InvalidKeyException, SignatureException {
+
+        final Signature signature =
+                Signature.getInstance(SignatureType.RSA.signingAlgorithm(), SignatureType.RSA.provider());
+
+        signature.initSign(keyPair.getPrivate());
+        signature.update(data);
+
+        return signature.sign();
+    }
+
+    /**
+     * Loads a pfx key from file and returns the key pair
+     *
+     * @param keyFileName a pfx key file
+     * @param password    the password for the key file
+     * @param alias       alias of the key
+     * @return a KeyPair
+     */
+    @NonNull
+    public static KeyPair loadPfxKey(
+            @NonNull final String keyFileName, @NonNull final String password, @NonNull final String alias) {
+
+        Objects.requireNonNull(keyFileName, "keyFileName");
+        Objects.requireNonNull(password, "password");
+        Objects.requireNonNull(alias, "alias");
+
+        try (final FileInputStream inputStream = new FileInputStream(keyFileName)) {
+            final KeyStore keyStore = KeyStore.getInstance(KEYSTORE_TYPE);
+            keyStore.load(inputStream, password.toCharArray());
+
+            return new KeyPair(keyStore.getCertificate(alias).getPublicKey(), (PrivateKey)
+                    keyStore.getKey(alias, password.toCharArray()));
+        } catch (final NoSuchAlgorithmException
+                | KeyStoreException
+                | UnrecoverableKeyException
+                | IOException
+                | CertificateException e) {
+
+            throw new RuntimeException("Unable to load Pfx key from file: " + keyFileName, e);
+        }
+    }
+
+    /**
+     * Sets up the constructable registry, and configures {@link SettingsCommon}
+     * <p>
+     * Should be called before using stream utilities
+     */
+    public static void initializeSystem() {
+        BootstrapUtils.setupConstructableRegistry();
+
+        // we don't want deserialization to fail based on any of these settings
+        SettingsCommon.maxTransactionCountPerEvent = Integer.MAX_VALUE;
+        SettingsCommon.maxTransactionBytesPerEvent = Integer.MAX_VALUE;
+        SettingsCommon.transactionMaxBytes = Integer.MAX_VALUE;
+        SettingsCommon.maxAddressSizeAllowed = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Generates a signature file for the given stream file
+     *
+     * @param destinationDirectory the directory to which the signature file will be saved
+     * @param streamType           type of the stream file
+     * @param streamFileToSign     the stream file to be signed
+     * @param keyPair              the keyPair used for signing
+     */
+    public static void signStreamFile(
+            @NonNull final File destinationDirectory,
+            @NonNull final StreamType streamType,
```

Comment:
In this class, this should always be `EventStreamType`, so this variable is not necessary